cmd/main.go

```diff
@@ -54,18 +62,24 @@ func main() {
 
 	names := []string{"init", "apply", "preview", "diff", "destroy", "status"}
 	initCmd := initcmd.NewCmdInit(f, ioStreams)
+	initCmd.PreRunE = preRunE
 	updateHelp(names, initCmd)
 	loader := manifestreader.NewManifestLoader(f)
 	invFactory := inventory.ClusterClientFactory{StatusPolicy: inventory.StatusPolicyNone}
 	applyCmd := apply.Command(f, invFactory, loader, ioStreams)
+	applyCmd.PreRunE = preRunE
 	updateHelp(names, applyCmd)
 	previewCmd := preview.Command(f, invFactory, loader, ioStreams)
+	previewCmd.PreRunE = preRunE
 	updateHelp(names, previewCmd)
 	diffCmd := diff.NewCommand(f, ioStreams)
+	diffCmd.PreRunE = preRunE
 	updateHelp(names, diffCmd)
 	destroyCmd := destroy.Command(f, invFactory, loader, ioStreams)
+	destroyCmd.PreRunE = preRunE
 	updateHelp(names, destroyCmd)
 	statusCmd := status.Command(f, invFactory, loader)
+	statusCmd.PreRunE = preRunE
```

Comment:
Doesn't cobra provide a way to register a preRunE function on all commands, rather than having to add them to each individually? Alternatively, could we just loop over all the commands and register it?

Reply:
> Doesn't cobra provide a way to register a preRunE function on all commands, rather than having to add them to each individually?

Yes, but we don't need it running for `help` or `completion`, since they don't talk tot he server.

> Alternatively, could we just loop over all the commands and register it?

Done.